tools/ramdump/ramdump_tool.c

```diff
@@ -332,25 +335,41 @@ int main(int argc, char *argv[])
 	int dev_fd;
 	int dev_lock_fd;
 	char *dev_file;
+	int port;
 
 	ret = 1;
 
 	if (argc < 2) {
 		printf("Usage: ./ramdump <device>\n");
-		printf("Ex   : ./ramdump /dev/ttyUSB1\n");
+		printf("Ex   : ./ramdump /dev/ttyUSB1   or\n");
+		printf("       ./ramdump /dev/ttyACM0\n");
 		return -1;
 	}
 
 	dev_file = argv[1];
 
-	if (access("/var/lock/LCK..ttyUSB1", F_OK) == 0) {
-		printf("Error : couldnt lock serial port device %s\n", "/dev/ttyUSB1");
-		printf("Please close any other process using this port first\n");
-		goto lock_check_err;
+	if (!strcmp(dev_file, "/dev/ttyUSB1")) {
+		port = ttyUSB1;
+	} else if (!strcmp(dev_file, "/dev/ttyACM0")) {
+		port = ttyACM0;
+	}
```

Comment:
How about below?
```
#define TTYPATH_LEN  25
#define TTYTYPE_LEN   6

char tty_path[TTYPATH_LEN] = { 0, };
char tty_type[TTYTYPE_LEN] = { 0, };

// Get the tty type
if (!strcmp(dev_file, "/dev/ttyUSB1")) {
	strncpy(tty_type, "ttyUSB1", TTYTYPE_LEN);
} else if (!strcmp(dev_file, "/dev/ttyACM0")) {
	strncpy(tty_type, "ttyACM0", TTYTYPE_LEN);
} else {
	printf("Undefined tty %s\n", dev_file);
	return -1;
}

// make the tty path
snprintf(tty_path, TTYPATH_LEN, "/var/lock/LCK..%s", tty_type);

if (access(tty_path, F_PK) == 0) {
 ...
}
...
if ((dev_lock_fd = open(tty_path, O_CREAT, S_IRWXU)) == -1) {
	printf("tty lock  failed\n");
	goto tty_lock_err;
}
...
remove(tty_path);
```

Reply:
This is a better method, have updated the logic with this approach.
Thank you.